src/distributions/normal.jl

```diff
@@ -538,37 +538,45 @@ end
 
 # Thes functions extends the `CVI` approximation method in case if input is from the `NormalDistributionsFamily`
 
-get_df_m(::Type{<:UnivariateNormalNaturalParameters}, ::Type{<:UnivariateGaussianDistributionsFamily}, logp_nc::Function) = (z) -> ForwardDiff.derivative(logp_nc, z)
-
-get_df_m(::Type{<:MultivariateNormalNaturalParameters}, ::Type{<:MultivariateGaussianDistributionsFamily}, logp_nc::Function) = (z) -> ForwardDiff.gradient(logp_nc, z)
+function compute_second_derivative(grad::G, logp::F, z_s::Real) where {G, F}
+    first_derivative = (x) -> compute_derivative(grad, logp, x)
+    return compute_derivative(grad, first_derivative, z_s)
+end
 
-get_df_v(::Type{<:UnivariateNormalNaturalParameters}, ::Type{<:UnivariateGaussianDistributionsFamily}, df_m::Function) = (z) -> ForwardDiff.derivative(df_m, z)
+function compute_df_mv(approximation::CVI, logp::F, z_s::Real) where {F}
+    df_m = compute_derivative(get_grad(approximation), logp, z_s)
+    df_v = compute_second_derivative(get_grad(approximation), logp, z_s)
+    return df_m, df_v / 2
+end
 
-get_df_v(::Type{<:MultivariateNormalNaturalParameters}, ::Type{<:MultivariateGaussianDistributionsFamily}, df_m::Function) = (z) -> ForwardDiff.jacobian(df_m, z)
+function compute_df_mv(approximation::CVI, logp_nc::F, z_s::Vector) where {F}
+    df_m = compute_gradient(get_grad(approximation), logp_nc, z_s)
+    df_v = compute_hessian(get_grad(approximation), logp_nc, z_s)
+    return df_m, df_v ./ 2
+end
 
-function render_cvi(approximation::CVIApproximation, logp_nc::F, initial::GaussianDistributionsFamily) where {F}
-    η = naturalparams(initial)
-    λ = naturalparams(initial)
+function prod(approximation::CVI, logp::F, dist::GaussianDistributionsFamily) where {F <: Function}
```

Comment:
I like this more and it also matches better with the already existing functionality. But 😅, the `prod` function in ReactiveMP currently expects a `pdf` (or a distribution), not `logpdf` as its argument. I'm making this comment for myself, no need to change anything @Nimrais , I'll deal with that discrepancy later myself during the final review.